src/Umbraco.Web/Search/SearchableTreeAttribute.cs

```diff
@@ -8,42 +7,47 @@ public sealed class SearchableTreeAttribute : Attribute
+        /// <param name="serviceName">Name of the service.</param>
+        public SearchableTreeAttribute(string serviceName)
+            : this(serviceName, string.Empty)
+        { }
 
         /// <summary>
-        /// This constructor defines both the angular service and method name to use and explicitly defines a sort order for the results
+        /// This constructor defines both the Angular service and method name to use.
         /// </summary>
-        /// <param name="serviceName"></param>
-        /// <param name="methodName"></param>
-        /// <param name="sortOrder"></param>
-        public SearchableTreeAttribute(string serviceName, string methodName, int sortOrder)
-        {
-            if (string.IsNullOrWhiteSpace(serviceName)) throw new ArgumentNullOrEmptyException(nameof(serviceName));
-            if (string.IsNullOrWhiteSpace(methodName)) throw new ArgumentNullOrEmptyException(nameof(methodName));
-            MethodName = methodName;
-            SortOrder = sortOrder;
-            ServiceName = serviceName;
-        }
+        /// <param name="serviceName">Name of the service.</param>
+        /// <param name="methodName">Name of the method.</param>
+        public SearchableTreeAttribute(string serviceName, string methodName)
+            : this(serviceName, methodName, DefaultSortOrder)
+        { }
 
         /// <summary>
-        /// This constructor will assume that the method name equals `format(searchResult, appAlias, treeAlias)`
+        /// This constructor defines both the Angular service and method name to use and explicitly defines a sort order for the results
         /// </summary>
-        /// <param name="serviceName"></param>
-        public SearchableTreeAttribute(string serviceName)
+        /// <param name="serviceName">Name of the service.</param>
+        /// <param name="methodName">Name of the method.</param>
+        /// <param name="sortOrder">The sort order.</param>
+        /// <exception cref="ArgumentNullException">serviceName
+        /// or
+        /// methodName</exception>
+        /// <exception cref="ArgumentException">Value can't be empty or consist only of white-space characters. - serviceName</exception>
+        public SearchableTreeAttribute(string serviceName, string methodName, int sortOrder)
         {
-            if (string.IsNullOrWhiteSpace(serviceName)) throw new ArgumentNullOrEmptyException(nameof(serviceName));
-            MethodName = "";
+            if (serviceName == null) throw new ArgumentNullException(nameof(serviceName));
+            if (string.IsNullOrWhiteSpace(serviceName)) throw new ArgumentException("Value can't be empty or consist only of white-space characters.", nameof(serviceName));
+
             ServiceName = serviceName;
-            SortOrder = DefaultSortOrder;
+            MethodName = methodName ?? throw new ArgumentNullException(nameof(methodName));
```

Comment:
We should allow an empty `methodName`, as the constructor without this argument sets it to an empty string anyway.

Reply:
but when using this overload, shouldn't we assume it's being used because the methodName should be supplied?

Reply:
Hmm, because it's an attribute, the arguments are quite static (besides dynamically emitting/adding attributes 🤡), so the overload shouldn't be used with an empty method name indeed.

But why add the additional check if the value is valid anyway and exceptions thrown in attributes should be avoided in general (as it gets thrown when getting the attribute)?